child_sync_typo3/model/child_compassion.py

```diff
@@ -0,0 +1,176 @@
+import base64
+
+from openerp.osv import orm
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+
+from sync_typo3 import Sync_typo3
+from datetime import datetime, timedelta
+
+
+class compassion_child(orm.Model):
+    _inherit = 'compassion.child'
+
+    def _get_typo3_child_id(self, cr, uid, child_code):
+        res_query = Sync_typo3.request_to_typo3(
+            "select * "
+            "from tx_drechildpoolmanagement_domain_model_children "
+            "where child_key='%s';" % child_code, 'sel')
+        res = 0
+        try:
+            res = json.loads(res_query)[0]['uid']
+        except:
+            raise orm.except_orm(
+                _('Typo3 Error'),
+                _('Child %s not found on typo3') % child_code)
+
+        return res
+
+    def child_add_to_typo3(self, cr, uid, ids, context=None):
+        self._recompute_unsponsored(cr, uid, ids, context)
+
+        # Solve the encoding problems on child's descriptions
+        reload(sys)
+        sys.setdefaultencoding('UTF8')
+
+        for child in self.browse(cr, uid, ids, context):
+            project_obj = self.pool.get('compassion.project')
+            project = project_obj.get_project_from_typo3(
+                cr, uid, child.project_id.code)
+
+            if not project:
+                project = project_obj.project_add_to_typo3(
+                    cr, uid, [child.project_id.id], context)[0]
+
+            child_gender = self._get_gender(cr, uid, child.gender, context)
+            child_image = child.code + "_f.jpg," + child.code + "_h.jpg"
+
+            today_ts = calendar.timegm(
+                datetime.today().utctimetuple())
+            consign_ts = timedelta(days=200).total_seconds()
+            if child.birthdate:
+                child_birth_date = calendar.timegm(
+                    datetime.strptime(child.birthdate, DF).utctimetuple())
+            else:
+                child_birth_date = 0
+            if child.unsponsored_since:
+                child_unsponsored_date = calendar.timegm(
+                    datetime.strptime(child.unsponsored_since,
+                                      DF).utctimetuple())
+            else:
+                child_unsponsored_date = today_ts
+
+            # Fix ' in description
+            child_desc_de = child.desc_de.replace('\'', '\'\'')
+            child_desc_fr = child.desc_fr.replace('\'', '\'\'')
+
+            # German description (parent)
+            Sync_typo3.request_to_typo3(
+                "insert into "
+                "tx_drechildpoolmanagement_domain_model_children"
+                "(child_key, child_name_full, child_name_personal,"
+                " child_gender, child_biography,"
+                " consignment_date, tstamp, crdate, consignment_expiry_date,"
+                " l10n_parent,image,child_birth_date,"
+                " child_unsponsored_since_date,project) "
+                "values ('{}','{}','{}','{}','{}','{}',"

```

Comment:
Can you take advantage of this pull request to change this and add indexes ({0} {1} {2}, etc...)